packages/viewer/src/lib/types/depthBuffer.ts

```diff
@@ -216,15 +217,33 @@ export class DepthBuffer implements FrameImageLike {
   public isOccluded(worldPt: Vector3.Vector3, viewport: Viewport): boolean {
     const { position, direction, projectionViewMatrix } = this.camera;
 
+    // Calculate the distance from the camera to the given world point
     const eyeToPoint = Vector3.subtract(worldPt, position);
-    const projected = Vector3.project(eyeToPoint, direction);
-    const distance = Vector3.magnitude(projected);
+    const distanceToPoint = Math.abs(Vector3.dot(eyeToPoint, direction));
 
-    const ndc = Vector3.transformMatrix(worldPt, projectionViewMatrix);
-    const screenPt = viewport.transformVectorToViewport(ndc);
+    // Find the screen point corresponding to the world point for the current camera
+    const screenPt = viewport.transformWorldToViewport(
+      worldPt,
+      projectionViewMatrix
+    );
     const scaledPt = viewport.transformPointToFrame(screenPt, this);
-    const depth = this.getLinearDepthAtPoint(scaledPt);
 
-    return distance > depth;
+    // Find the depth of the closest geometry at the same point on the screen
+    // Use the correct calculation for the camera type
+    const isPerspectiveCamera = this.camera.isPerspective();
+    const depthOfClosestGeometry = isPerspectiveCamera
+      ? this.getLinearDepthAtPoint(scaledPt)
+      : this.getOrthographicDepthAtPoint(scaledPt);
+
+    // Allow for a small rounding error
+    // Note that if the world point is coincident with the geometry,
+    // we want to err on the side of returning not occluded
+    const allowableDifference = Math.abs(0.02 * distanceToPoint);
```

Comment:
@mikemccarty-vertex noticed this, but assuming the math above is the same, I don't think this will apply a uniform tolerance across all points. In particular, the tolerance should be adjusted the further you move from the center of the viewport.

Reply:
Ideally, the tolerance is a function of something like `far - near` -- i.e. one that is constant for all pins in the view.

Reply:
Updated!

Reply:
Actually, while I agree that it we should look into setting the tolerance such that it is constant for all pins in the view, I don't like the current results we're getting after changing it to a function of (far-near), so I reverted back to using a function of the distanceToPoint. Further work here can be included in the follow-up work.